designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,284 @@
+| Authors       | Creation Date | Status      | Extra |
+|---------------|---------------|-------------|---|
+| @dashanji,@camilamacedo86,@LCaparelli... | Sep, 2023 | Implementeble | - |
+
+# Allow users package Operator with Helm Charts
+
+This proposal aims to introduce an optional mechanism that allows users to generate a Helm Chart from their Kubebuilder-scaffolded project. This will enable them to effectively package and distribute their solutions.
+
+To achieve this goal, we can introduce a Helm Chart plugin (i.e. helm-chart/v1alpha), which will provide the scaffolds. Alternatively, we could implement a command to generate the Helm Chart from a provided Kubebuilder project.
+
+## Example
+
+**Via a new Optional Plugin**
+
+There are several use cases for this plugin
+as follows:
+
+#### Case 1: Initial Helm chart distribution
+
+For users who have developed an Operator and wish to distribute it via a Helm chart for the first time:
+
+By running `kubebuilder init|edit --plugins=helm-chart/v1alpha`, the scaffolds of Helm Chart to package the Operator would be created in the project as follows:
+
+```shell
+$ tree
+.
+..
+├── PROJECT
+...
+├── deploy
+│   └── helm-chart // The helm chart with all values will be scaffold under a directory of the project
+...
+```
+
+The initial setup will be straightforward, creating the Helm chart structure and populating it with the default values as determined by the project's configuration.
+
+#### Case 2: Update the Helm chart with manifests changes
+
+For users who want to update an existing Helm chart to reflect changes in the kustomize directory, such as CRDs, Webhooks, or RBAC roles:
+
+The plugin will add specific targets within the Makefile that are tailored for Helm chart scenarios. When you run `make manifests` or `make generate` commands, CRDs, RBAC roles and webhooks will be automatically synchronized with the Helm chart.
```

Comment:
I think we need to generate the manifests when we run the plugin
all code regards this feature will be implemented the plugin so we can assume that only the helm-chart new plugin knows how to transform kubebuilder project files to helm-chart values

Reply:
Agree, in this way, users only need to care about how to use this helm-chart plugin, and the more fine-grained control can be optimized according to needs later.